test-three/amy-menu/src/menu.tsx

```diff
@@ -0,0 +1,93 @@
+import { Component } from "react";
+import menuJson from "./menu-data.json";
+
+type MenuItemProps = {
+  name: string;
+};
+
+class MenuSubItem extends Component<MenuItemProps> {
+  render() {
+    return (
+      <li>
+        {this.props.name}
+      </li>
+    );
+  }
+}
+
+class MenuItem extends Component<MenuItemProps> {
+  render() {
+    return (
```

Comment:
While not mentioned in the test do you think you'd be able to modify your menu to contain accordions so that the top level items expand/collapse to show/hide the child items?

eg:

![image](https://user-images.githubusercontent.com/6108586/120059013-12512c80-c092-11eb-8e3d-a8a08286b3b8.png)

Reply:
For the menu to contain accordian functionality, I would implement state for a boolean called expandMenuItems. Because we want to use a functional component, a state hook would be used so we don’t have to write a stateful component. This expandMenuItems would only be used for menuItems with children.